src/utils/pools.ts

```diff
@@ -126,12 +127,30 @@ export async function getPoolReserves(poolContract: Contracts, chainId: ChainId,
       };
     } else if (isVault) {
       console.log(`isVault`);
-      // vaults
-      let reserveBalances = await asIchiVault(poolContract).getTotalAmounts();
-      return {
-        _reserve0: Number(reserveBalances.total0),
-        _reserve1: Number(reserveBalances.total1)
-      };
+
+      const ichiVaultInstance = asIchiVault(poolContract);
+      const exceptionAddress = VAULTS[chainId].ICHI.address;
+      const provider = await getProvider(chainId);
+      if (!provider) {
+        throw new Error(`Could not get provider`);
+      }
+
+      if (ichiVaultInstance.address == exceptionAddress) {
+        let ichiTokenContract = getErc20Contract(TokenName.ICHI_V2, provider);
+        let reserveBalances = await ichiVaultInstance.getBasePosition();
+        let contractBalance = await ichiTokenContract.balanceOf(exceptionAddress);
```

Comment:
Combine these two so they run in parallel.   Something like `let [result1,  result2] = await Promise.all([promiseOne(), promiseTwo()])`